src/user-guide/writing-workflows/optional-outputs.rst

```diff
@@ -0,0 +1,266 @@
+
+Leaf tasks (with nothing downstream of them) can have optional outputs. The
+following workflow can complete successfully whether ``bar`` succeeds or fails:
+
+.. code-block:: cylc
+
+   [graph]
+       R1 = "foo => bar?"
+
+.. note::
+   
+   Optionality is an attribute of the upstream task output, not a triggering
+   condition. The trigger ``foo? => bar`` says to *trigger ``bar`` if ``foo``
+   succeeds*, and the ``?`` just tells the scheduler not to retain ``foo`` as
+   an incomplete task if it fails.
+
+
+Finish Triggers
+---------------
+
+Task ``:finish`` is really a pseudo output used as shorthand to triggger off
+either of the real ``:succeed`` or ``:fail`` outputs. Consequently use of a
+``foo:finish`` (say) implies that ``foo:succeed`` (or ``foo:fail``) must be
+marked as optional if it occurs anywhere in the graph. And ``:finish?`` is
+illegal syntax because is would incorreclty imply that "finishing is optional".
+
+.. code-block:: cylc
+
+   [graph]
+       R1 = """
+          foo:finish => bar
+          foo? => baz  # must be optional!
+       """
+
+
+Family Triggers
+---------------
+
+.. (taken from https://github.com/cylc/cylc-flow/pull/4343#issuecomment-913901972)
+
+Family triggers such as ``FAM:succeed-all`` and ``FAM:fail-any`` do not refer
+to real "family outputs", they are short for logical combinations of member
+task outputs.
+
+As such, family triggers do not dictate the optionality of member outputs.
+However, they are often used without separate reference to specific members in
+the graph so they do imply a default optionality for member outputs:
+
+- All family triggers imply (by default) that the corresponding member outputs
+  are *expected*
+
+  - Unless members are singled out as *optional* elsewhere in the graph
+  - Except for family ``:finish-all,any`` triggers, which like
+    task ``:finish`` triggers imply optional outputs
+- The default can be changed by using ``?`` on the family trigger
+
+Examples below are for a family ``A`` with member tasks ``a<i>``
+for ``i = 1, 2, 3``; family ``B`` with members ``b<i>``; and so on.
+Results are the same if ``-all`` is replaced by ``-any`` on all family triggers.
+
+.. code-block:: cylc
+
+   [graph]
+       R1 = """
+          A:succeed-all => x1  # a<i>:succeed expected
+          B:succeed-all? => x2  # b<i>:succeed optional
+
+          C:fail-all => x3  # c<i>:fail expected
+          D:fail-all? => x4  # d<i>:fail optional
+
+          E:succeed-all => x5  # e<i>:succeed expected ...
+          e2? => z5            # ... but e2:succeed optional
+
+          F:finish-all => x6  # f<i>:succeed optional
+          F:finish-all? => x6  # ERROR (c.f. task:finish rules)
+       """
+
+
+.. _path-branching:
+
+Alternate Path Branching
+------------------------
+
+The graph splits into concurrent branches whenever several tasks trigger off of a
+single upstream parent:
+
+.. code-block:: cylc
+
+   [graph]
+       R1 = "foo => bar & baz"
```

Comment:
From past experience graph branching causes a lot of confusion. I think diagrams help a lot in clarification, the old suicide trigger tutorial had diagrams which can easily be repurposed here.

Here are my attempts as suggestions (please check!):

```suggestion
   [graph]
       R1 = "foo => bar & baz"

.. digraph:: Example
   :align: center

   foo -> bar
   foo -> baz
```